maven-wrapper-distribution/src/resources/only-mvnw

```diff
@@ -70,11 +70,19 @@ set_java_home() {
 
 # hash string like Java String::hashCode
 hash_string() {
-  str="${1:-}" h=0
-  while [ -n "$str" ]; do
-    h=$(( ( h * 31 + $(LC_CTYPE=C printf %d "'$str") ) % 4294967296 ))
-    str="${str#?}"
+  str="${1}"
+  length=${#str}
+  h=0
+  # strings start with index 1
+  idx=1
+
+  # lt: skip line break at end
+  while [ $idx -lt "$length" ]; do
+    char=$( printf "%s" "$str" | cut -c $idx )
```

Comment:
Fork here, alternative is to use string index (supported by ksh and bash and zsh). However:

```
In maven-wrapper-distribution/src/resources/only-mvnw line 80:
    char=${str:${idx}:1}
         ^-------------^ SC3057 (warning): In POSIX sh, string indexing is undefined.
```

Reply:
The other advantage is that string index starts at 0 😄

Reply:
According to Ubuntu, they still use Dash as default symlink target of `/bin/sh` which does not know about variable substitution: https://wiki.ubuntu.com/DashAsBinSh

They recommend using awk (but cut is enough here)